src/main/java/com/minecolonies/coremod/entity/ai/util/RecipeStorage.java

```diff
@@ -49,6 +80,25 @@ public RecipeStorage(final List<ItemStack> input, final int gridSize, final Item
         this.primaryOutput = primaryOutput;
         this.secondaryOutput = Collections.unmodifiableList(Arrays.asList(secondaryOutput));
         this.gridSize = gridSize;
+        this.intermediate = null;
+    }
+
+    /**
+     * Create an instance of the recipe storage.
+     *
+     * @param input           the list of input items (required for the recipe).
+     * @param gridSize        the required grid size to make it.
+     * @param primaryOutput   the primary output of the recipe.
+     * @param intermediate    the intermediate to use (e.g furnace).
+     * @param secondaryOutput the secondary output (like buckets or similar).
+     */
+    public RecipeStorage(final List<ItemStack> input, final int gridSize, final ItemStack primaryOutput, final Block intermediate, final ItemStack... secondaryOutput)
```

Comment:
Secondary Outputs can be determined by calling a method on the ItemStack itself.